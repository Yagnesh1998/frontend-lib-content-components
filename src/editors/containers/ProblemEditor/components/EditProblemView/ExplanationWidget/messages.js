export const messages = {
  solutionWidgetTitle: {
    id: 'authoring.problemEditor.explanationwidget.explanationWidgetTitle',
    defaultMessage: 'Explanation',
    description: 'Explanation Title',
  },
  solutionDescriptionText: {
    id: 'authoring.problemEditor.solutionwidget.solutionDescriptionText',
    defaultMessage: 'Provide an explantion for the correct answer',
    description: 'Description of the solution widget',
  },
  placeholder: {
    id: 'authoring.problemEditor.questionwidget.placeholder',
    defaultMessage: 'Enter your explanation',
    description: 'Placeholder text for tinyMCE editor',
  },
};

export default messages;
