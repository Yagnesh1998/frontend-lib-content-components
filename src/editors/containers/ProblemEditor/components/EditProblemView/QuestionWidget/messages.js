export const messages = {
  questionWidgetTitle: {
    id: 'authoring.questionwidget.question.questionWidgetTitle',
    defaultMessage: 'Question',
    description: 'Question Title',
  },
  placeholder: {
    id: 'authoring.problemEditor.questionwidget.placeholder',
    defaultMessage: 'Enter your question',
    description: 'Placeholder text for tinyMCE editor',
  },
};

export default messages;
