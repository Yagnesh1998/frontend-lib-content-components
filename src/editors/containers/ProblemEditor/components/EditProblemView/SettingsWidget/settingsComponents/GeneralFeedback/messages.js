export const messages = {
  generalFeebackSettingTitle: {
    id: 'authoring.problemeditor.settings.generalFeebackSettingTitle',
    defaultMessage: 'General Feedback',
    description: 'label for general feedback setting',
  },
  generalFeedbackInputLabel: {
    id: 'authoring.problemeditor.settings.generalFeedbackInputLabel',
    defaultMessage: 'Enter the feedback to appear when a student submits a wrong answer. This will be overrdidden if you add answer-specific feedback.',
    description: 'label for general feedback input describing rules',
  },
  generalFeedbackDescription: {
    id: 'authoring.problemeditor.settings.generalFeedbackInputDescription',
    defaultMessage: 'Enter the feedback to appear when a student submits a wrong answer. This will be overrdidden if you add answer-specific feedback.',
    description: 'description for general feedback input, clariying useage',
  },
};
export default messages;
