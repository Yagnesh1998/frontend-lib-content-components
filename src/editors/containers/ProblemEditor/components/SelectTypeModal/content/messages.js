export const messages = {
  previewTitle: {
    id: 'authoring.problemEditor.preview.title',
    defaultMessage: '{previewTitle}',
    description: 'Title for the problem preview column',
  },
  previewAltText: {
    id: 'authoring.problemEditor.preview.altText',
    defaultMessage: `A preview illustration of a {problemType, select,
      multiplechoiceresponse {single select}
      stringreponse {text input}
      numericalresponse {numerical input}
      optionresponse {dropdown}
      choiceresponse {multiple select}
      other {null}
    } problem`,
    description: 'Alt text for the illustration of the problem preview',
  },
  previewDescription: {
    id: 'authoring.problemEditor.preview.description',
    defaultMessage: '{previewDescription}',
    description: 'Description of the selected problem type',
  },
  learnMoreButtonLabel: {
    id: 'authoring.problemEditor.learnMoreButtonLabel.label',
    defaultMessage: 'Learn More',
    description: 'Label for Learn More button',
  },
};

export default messages;
