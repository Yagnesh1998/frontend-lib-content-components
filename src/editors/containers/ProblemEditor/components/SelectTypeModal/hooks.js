import {
  useState,
} from 'react';
import { StrictDict } from '../../../../utils';

export const state = StrictDict({
  selected: (val) => useState(val),
});

export const onSelect = ({ selected }) => {
  return;
};

export default { state, onSelect };
