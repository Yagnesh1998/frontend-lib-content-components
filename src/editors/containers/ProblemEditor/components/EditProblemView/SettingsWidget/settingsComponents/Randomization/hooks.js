import { useState, useEffect } from 'react';
import messages from './messages';
import { RandomizationTypes } from '../../../../../../../data/constants/problem';
import * as module from './hooks';

export const state = {
  summary: (val) => useState(val),
};

export const useRandomizationSettingStatus = ({ randomization, updateSettings }) => {
  const [summary, setSummary] = module.state.summary({ message: messages.noRandomizationSummary, values: {} });
  useEffect(() => {
    setSummary({
      message: randomization ? RandomizationTypes[randomization] : messages.noRandomizationSummary,
    });
  }, [randomization]);

  const handleChange = (event) => {
    updateSettings({ randomization: event.target.value });
  };
  return { summary, handleChange };
};

export default useRandomizationSettingStatus;
