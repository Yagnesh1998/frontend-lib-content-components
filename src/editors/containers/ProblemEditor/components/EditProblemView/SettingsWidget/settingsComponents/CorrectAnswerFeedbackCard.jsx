import React from 'react';
import PropTypes from 'prop-types';

import { injectIntl, FormattedMessage, intlShape } from '@edx/frontend-platform/i18n';
import { Form } from '@edx/paragon';

import SettingsOption from '../SettingsOption';
import messages from '../messages';
import { correctAnswerFeedbackHooks } from '../hooks';

export const CorrectAnswerFeedbackCard = ({
  correctAnswerFeedback,
  updateSettings,
  // inject
  intl,
}) => {
  const { summary, handleChange } = correctAnswerFeedbackHooks(correctAnswerFeedback, updateSettings);
  return (
    <SettingsOption
      title={intl.formatMessage(messages.correctAnswerFeedbackSettingTitle)}
      summary={summary.intl ? intl.formatMessage(summary.message) : summary.message}
      none={!correctAnswerFeedback}
    >
      <div className="halfSpacedMessage">
        <span>
          <FormattedMessage {...messages.correctAnswerFeedbackSettingText} />
        </span>
      </div>
      <Form.Group>
        <Form.Control
          value={correctAnswerFeedback}
          onChange={handleChange}
          floatingLabel={intl.formatMessage(messages.correctAnswerFeedbackInputLabel)}
        />
      </Form.Group>
    </SettingsOption>
  );
};

CorrectAnswerFeedbackCard.propTypes = {
  correctAnswerFeedback: PropTypes.string.isRequired,
  updateSettings: PropTypes.func.isRequired,
  intl: intlShape.isRequired,
};

export default injectIntl(CorrectAnswerFeedbackCard);
