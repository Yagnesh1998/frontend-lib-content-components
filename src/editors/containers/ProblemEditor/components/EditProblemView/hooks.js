import 'tinymce';
import ReactStateSettingsParser from '../../data/ReactStateSettingsParser';
import ReactStateOLXParser from '../../data/ReactStateOLXParser';
import { setAssetToStaticUrl } from '../../../../sharedComponents/TinyMceWidget/hooks';

export const fetchEditorContent = () => {
  const editorObject = {};
  const EditorsArray = window.tinymce.editors;
  Object.entries(EditorsArray).forEach(([id, editor]) => {
    if (Number.isNaN(parseInt(id))) {
      if (id.startsWith('answer')) {
        const { answers } = editorObject;
        const answerId = id.substring(id.indexOf('-') + 1);
        editorObject.answers = { ...answers, [answerId]: editor.getContent() };
      } else {
        editorObject[id] = editor.getContent();
      }
    }
  });
  return editorObject;
};

export const parseState = ({
  problem,
  isAdvanced,
  ref,
  assets,
  lmsEndpointUrl,
}) => () => {
  const editorObject = fetchEditorContent();
  const reactSettingsParser = new ReactStateSettingsParser(problem);
  const reactOLXParser = new ReactStateOLXParser({ problem, editorObject });
  const reactBuiltOlx = setAssetToStaticUrl({ editorValue: reactOLXParser.buildOLX(), assets, lmsEndpointUrl });
  const rawOLX = ref?.current?.state.doc.toString();
  return {
    settings: reactSettingsParser.getSettings(),
    olx: isAdvanced ? rawOLX : reactBuiltOlx,
  };
};
