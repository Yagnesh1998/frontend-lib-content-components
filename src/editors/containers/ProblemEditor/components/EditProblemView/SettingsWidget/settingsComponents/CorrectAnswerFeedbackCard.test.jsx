import React from 'react';
import { shallow } from 'enzyme';
import { formatMessage } from '../../../../../../../testUtils';
import { correctAnswerFeedbackHooks } from '../hooks';
import { CorrectAnswerFeedbackCard } from './CorrectAnswerFeedbackCard';
import messages from '../messages';

jest.mock('../hooks', () => ({
  correctAnswerFeedbackHooks: jest.fn(),
}));

describe('CorrectAnswerFeedbackCard', () => {
  const correctAnswerFeedback = 'SomEFeEdbacK';
  const props = {
    correctAnswerFeedback,
    updateSettings: jest.fn().mockName('args.updateSettings'),
    intl: { formatMessage },
  };

  describe('behavior', () => {
    it(' calls correctAnswerFeedbackHooks when initialized', () => {
      const correctAnswerFeedbackProps = {
        summary: { message: correctAnswerFeedback, values: {}, intl: false },
        handleChange: jest.fn().mockName('correctAnswerFeedbackHooks.handleChange'),
      };
      correctAnswerFeedbackHooks.mockReturnValue(correctAnswerFeedbackProps);
      shallow(<CorrectAnswerFeedbackCard {...props} />);
      expect(correctAnswerFeedbackHooks).toHaveBeenCalledWith(correctAnswerFeedback, props.updateSettings);
    });
  });

  describe('snapshot', () => {
    test('snapshot: renders correct answer feedback setting card', () => {
      const correctAnswerFeedbackProps = {
        summary: { message: correctAnswerFeedback, values: {}, intl: false },
        handleChange: jest.fn().mockName('correctAnswerFeedbackHooks.handleChange'),
      };
      correctAnswerFeedbackHooks.mockReturnValue(correctAnswerFeedbackProps);
      expect(shallow(<CorrectAnswerFeedbackCard {...props} />)).toMatchSnapshot();
    });
    test('snapshot: renders correct answer feedback setting card no key', () => {
      const correctAnswerFeedbackProps = {
        summary: { message: messages.noCorrectAnswerFeedbackSummary, values: {}, intl: true },
        handleChange: jest.fn().mockName('correctAnswerFeedbackHooks.handleChange'),
      };
      correctAnswerFeedbackHooks.mockReturnValue(correctAnswerFeedbackProps);
      expect(shallow(<CorrectAnswerFeedbackCard {...props} correctAnswerFeedback="" />)).toMatchSnapshot();
    });
  });
});
