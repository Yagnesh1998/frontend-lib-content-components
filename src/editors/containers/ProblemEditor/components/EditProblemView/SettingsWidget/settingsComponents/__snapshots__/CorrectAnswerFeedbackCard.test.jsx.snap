// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CorrectAnswerFeedbackCard snapshot snapshot: renders correct answer feedback setting card 1`] = `
<SettingsOption
  className=""
  extraSections={Array []}
  none={false}
  summary="SomEFeEdbacK"
  title="Correct Answer Feedback"
>
  <div
    className="halfSpacedMessage"
  >
    <span>
      <FormattedMessage
        defaultMessage="Enter the feedback to appear when a student submits a correct answer."
        description="Correct Answer Feedback settings card text"
        id="authoring.problemeditor.settings.correctAnswerFeedback.text"
      />
    </span>
  </div>
  <Form.Group>
    <Form.Control
      floatingLabel="Feedback message"
      onChange={[MockFunction correctAnswerFeedbackHooks.handleChange]}
      value="SomEFeEdbacK"
    />
  </Form.Group>
</SettingsOption>
`;

exports[`CorrectAnswerFeedbackCard snapshot snapshot: renders correct answer feedback setting card no key 1`] = `
<SettingsOption
  className=""
  extraSections={Array []}
  none={true}
  summary="None"
  title="Correct Answer Feedback"
>
  <div
    className="halfSpacedMessage"
  >
    <span>
      <FormattedMessage
        defaultMessage="Enter the feedback to appear when a student submits a correct answer."
        description="Correct Answer Feedback settings card text"
        id="authoring.problemeditor.settings.correctAnswerFeedback.text"
      />
    </span>
  </div>
  <Form.Group>
    <Form.Control
      floatingLabel="Feedback message"
      onChange={[MockFunction correctAnswerFeedbackHooks.handleChange]}
      value=""
    />
  </Form.Group>
</SettingsOption>
`;
